### PR TITLE
ath79: add support for ELECOM WAB-I1750-PS

### PR DESCRIPTION
**Pull request by zpc0**
ELECOM WAB-I1750-PS is an enterprise AP based on Edimax WAP1750.

Specifications
--
- SoC: Qualcomm Atheros QCA9558
- RAM: 128MiB, Winbound W9751G6KB251 x2
- Flash memory: 16MiB, MXIC MX25L1283FMI
- Wi-Fi: Qualcomm Atheros QCA9558
- Wi-Fi: Qualcomm Atheros QCA9880-BR4A
- Ethernet eth0: Atheros AR8035, PoE IN
- Ethernet eth1: Atheros AR8033, PoE OUT
- LED: 7x green LEDs, 1x red LEDs (3x gpio controlled LEDs)
- USB: 1x USB 2.0
- Input: 3x tactile switches
- Buzzer: 1x gpio controlled piezo buzzer
- Serial console: Cisco-compatible "SERIAL" port, TTL-level "SERVICE" port
- Power: DC 12V or PoE IN

About HSUART (high-speed UART)
--
QCA9558 has two UART interfaces. The first is 16550 compatible
low-speed UART. The second is AR9330 UART compatible high-speed
UART. WAB-I1750-PS assigns low-speed UART to the "SERIAL" port
for user, and high-speed UART to the "SERVICE" port for debugging.

The WAB-I1750-PS U-Boot does not initialize necessary registers for
HSUART, so we need to set correct values in dts.

About SERVICE port
--
WARNING: SERVICE port is NOT "Cisco-compatible". Do not connect
such cables to the SERVICE port.

To use the SERVICE port, a customized LAN cable and USB TTL serial
converter are required.

Cut LAN cable, then connect one side to USB TTL serial converter.

`SERVICE port --- T568B connector --- LAN cable --- USB TTL serial converter`
`Pin 1: 3V3 -------------- White/Orange --------------- 3V3`
`Pin 2: GND -------------- Orange --------------------- GND`
`Pin 3: TXD -------------- White/Green ---------------- RXD`
`Pin 4: RXD -------------- Blue ----------------------- TXD`

Flash instructions
--
1. Open the firmware upgrade page in Web UI
2. Perform firmware upgrade using factory.bin

### CODE REVIEW

File: target/linux/ath79/dts/qca9558_edimax_wap1750.dtsi [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+
+		partitions {
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_edimax_wap1750.dtsi:44]: I don't like the splitting of the flash/partitions node at this level. If there are really so many differences, the full flash nodes should be kept closer to the devices.

**Comment by adschm** [comment]
The first SoB needs to match the From: field. So, the correct order for the third patch is actually:
````
Signed-off-by: Yanase Yuki <dev@zpc.sakura.ne.jp>
Co-developed-by: INAGAKI Hiroshi <musashino.open@gmail.com>
Signed-off-by: INAGAKI Hiroshi <musashino.open@gmail.com>
```

File: target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+			partition@f70000 {
+				label = "backup";
+				reg = <0x0f70000 0x0010000>;
+				read-only;
+			};
+
+			partition@f80000 {
+				label = "storage";
+				reg = <0x0f80000 0x0080000>;
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts:163]: Consider using 6 digits here as commonly done for SPI on other ath79 devices.

File: target/linux/ath79/dts/qca9558_edimax_wap1750.dtsi (hunk 1 above)
Comment (zpc0) [reply target/linux/ath79/dts/qca9558_edimax_wap1750.dtsi]: OK. Fixed.

File: target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts (hunk 2 above)
Comment (zpc0) [reply target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts]: Fixed.

**Comment by zpc0** [comment]
> The first SoB needs to match the From: field. So, the correct order for the third patch is actually:
> 
> ```
> Signed-off-by: Yanase Yuki <dev@zpc.sakura.ne.jp>
> Co-developed-by: INAGAKI Hiroshi <musashino.open@gmail.com>
> Signed-off-by: INAGAKI Hiroshi <musashino.open@gmail.com>
> ```

Thanks. Fixed.

**Comment by adschm** [comment]
I've picked the elx-header patch for now ...

File: target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	phy-mode = "sgmii";
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts:225]: That's default in qca955x.dtsi and can be dropped.

Comment (zpc0) [reply target/linux/ath79/dts/qca9558_elecom_wab-i1750-ps.dts]: Fixed.

**Comment by zpc0** [comment]
> I've picked the elx-header patch for now ...

Thank you.
I removed applied patch.

**Comment by zpc0** [comment]
Update:
- remove uart node (3a4b7511)
- add patches for Linux 5.10

**Comment by zpc0** [comment]
update
- rebase
- fix patch number conflicts

**Pull request closed without merging**